core/interpreter/interpreter.cpp

```diff
@@ -349,14 +349,34 @@ class Interpreter {
             return res;    
         }
 
+        InterpreterResult process_double_op(string token, shared_ptr<Node> node, InterpreterResult res, shared_ptr<Context> _context) {
+            shared_ptr<Node> previous_node = make_shared<Node>();
+            previous_node->set_type(NODE_ACCESS);
+            previous_node->set_token(node->token);
+
+            InterpreterResult previous_value;
+            
+            previous_value = res.register_result(visit_node(previous_node, _context));
+            if(res.should_return()) { return res; }
+            
+            shared_ptr<Token> _token = make_shared<Token>(token);
+            InterpreterResult new_value;
+            new_value = res.register_result(res.process_binary(previous_value, _token, visit_unary_node(node, _context)));
```

Comment:
<code>visit_unary_node(node, _context)</code> should be changed to a <code>right_result</code> and declaring the <code>right_result</code> somewhere before with value like this:

<code>
InterpreterResult right_result;
right_result.set_from(1);
</code>

Reply:
otherwise good job uwu

Reply:
uwu will fix!